### DISCUSSION TITLE
Container startup order

### DISCUSSION THREAD

**Discussion by Moorsy-AU** [louislam/dockge#669] [Ask for Help] [answered]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/dockge/security/policy)

### 📝 Describe your problem

Hi team,

i can see a feature request for startup order for containers. how are containers started when dockge is restarted ? is it alphabetical order ? is it possible to tell dockge the order containers should boot ? for instance id like my cloudflare, tailscale, proxy manager to boot first, then plex, then the remaining arrs. 
Or do i need to create a stack for each function ? then again how do i know  that stack will start up first ?

### 🐻 Dockge Version

1.4.2

### 💻 Operating System and Arch

TrueNAS-SCALE-24.04.2.5 via Sandbox (Jailmaker)

### 🌐 Browser

Edge

**Comment by mizady** [comment, marked as answer]
Dockge is a front-end for docker, Dockge does not actually start the containers or maintain their state, it only displays information from docker and allows you to send commands to docker.  

You can set the depends on of each container to have some control over start-up order.  One method you can use but it depends on the containers you're spinning up would be to use a healthcheck by setting a script to run in the container prior to the application starting (that's the part that depends on the container because some may require creating your own docker file to actually run the script prior to application startup).  Basically a bash script to check for a healthy status (use wget or curl assuming the container has a web based health check for example) and if it doesn't return healthy (or whatever the healthy return value is) then sleep 1; or 10 or whatever you want to use for the check frequency and put it in a loop based on that result.  if the container you're going to run has docker socket access then you can even use docker inspect to get the health status instead of wget or curl.  the only problem with this solution is that if the dependency goes down, then the script you have in place will not be of any use as it would need to be called again to begin waiting for the other service again.

if you can put all the services in the same compose file and use depends on, that will simplify the process greatly and increase reliability.

**Reply by Moorsy-AU** [reply]
Thanks for the clarification @mizady. I have been reading about the health check and depends on as you mentioned. Creating stacks and using depends on does seem the easiest option given my current level of knowledge. 
Back to spending quality time with my matea ChatGPT and Youtube. 
Thanks again for the response.

**Reply by mizady** [reply]
the documentation is a good source as well, usually a mix of that and chatgpt can get you pretty far. 

[https://docs.docker.com/compose/how-tos/startup-order/](https://docs.docker.com/compose/how-tos/startup-order/)
